crates/core/tedge_mapper/src/c8y/tests.rs

```diff
@@ -1124,6 +1124,149 @@ async fn mapper_handles_multiline_sm_requests() {
     c8y_mapper.abort();
 }
 
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+#[serial]
+async fn mapper_publishes_supported_operations() {
+    // The test assures tede-mapper reads/parses the operations from operations directory and
+    // correctly publishes the supported operations message on `c8y/s/us`
+    // and verifies the supported operations that are published by the tedge_mapper.
+    let broker = mqtt_tests::test_mqtt_broker();
+    let mut messages = broker.messages_published_on("c8y/s/us").await;
+
+    let (_temp_dir, _ops_dir, sm_mapper) = start_c8y_mapper(broker.port).await.unwrap();
+
+    publish_a_fake_jwt_token(broker).await;
+
+    // Expect smartrest message on `c8y/s/us` with expected payload "114,c8y_TestOp2,c8y_TestOp1"
+    mqtt_tests::assert_received_all_expected(
+        &mut messages,
+        TEST_TIMEOUT_MS,
+        &["114,c8y_TestOp2,c8y_TestOp1\n"],
+    )
+    .await;
+    sm_mapper.abort();
+}
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+#[serial]
+async fn mapper_publishes_child_device_create_message() {
+    // The test assures tedge-mapper checks if there is a directory for operations for child devices, then it reads and
```

Comment:
Found it now. The child device directory implicitly getting created as part of the generic `start_c8y_mapper` function is counter-intuitive. I'd rather explicitly create the child directories and operation files in the test itself so that it's more obvious for a reader of this test.

Reply:
Refactored the code to create the `Operations` in the test itself.